vic/src/plugins/ext_mpi/ext_mpi_support.c

```diff
@@ -488,6 +488,92 @@ mpi_map_decomp_domain_basin(size_t   ncells,
         }
     }
 
+    // set mapping array
+    for (i = 0, l = 0; i < (size_t) mpi_size; i++) {
+        for(j=0;j<basins.Nbasin;j++){
```

Comment:
Just a coding style thing: be more consistent with the use of spaces in your code. Especially in nested loops like this. In VIC we separate almost everything with spaces.